app-config.yaml

```diff
@@ -1,15 +1,11 @@
 app:
   title: Backstage Example App
-  baseUrl: http://localhost:3000
+  baseUrl: http://localhost:7000
```

Comment:
Yeah not super clear for me either, since we're suggesting running the backend with NODE_ENV=development in docker 😅 . In general this PR switches the base `app-config.yaml` to be the production configuration, where the `app-config.development.yaml` provides overrides for local development.

CORS is moved to local development only, since it's not needed when the app is bundled with the backend.